### ISSUE TITLE
Can't register macros for actions

### ISSUE THREAD

**Issue by KasparRosin** [laravel/nova-issues#3465]
Title basically says it all. 

Would like to create macros for actions.
Example use case: Hide action from specific relationship resource fields.
```
->hideFromRelation(User::class)
```

**Comment by davidhemphill** [comment]
👋 Hi there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by crynobone** [comment]
Can you share how the full example how above would works? This seem to be reasonable request but would like to verify with some sample usage first.

**Comment by KasparRosin** [comment]
I had something like this in my mind

```php
Action::macro('hideFromRelations', function ($class) {
     $this->canSeeCallback = fn ($request) => !in_array($request->viaResource, collect($class)->toArray());
     return $this;
});
```

This would allow me to re-use this functionality much more easily
```php
SomeAction::make()
    ->hideFromRelations([User:class, Post::class]) // Pass in array of classes
    ->hideFromRelations(User:class) // Pass in single class
```

**Comment by crynobone** [comment]
Submit the PR.

```php
use App\Nova\User;
use Laravel\Nova\Actions\Action;
use Laravel\Nova\Http\Requests\ActionRequest;

Action::macro('hideFromRelations', function ($classes) {
     return $this->canSee(function ($request) {
          if ($request instanceof ActionRequest) {
               return true;
          }

          return ! in_array($request->viaResource, collect($classes)->transform(function ($class) {
               return $class::uriKey();
          })->all());
    });
});
```

**Comment by KasparRosin** [comment]
Thank you! Looking forward to this feature being added.

**Comment by crynobone** [comment]
Fixed in v3.28.0: https://nova.laravel.com/releases/3.28.0
